bansa/src/test/kotlin/com/brianegan/bansa/ReselectTest.kt

```diff
@@ -0,0 +1,409 @@
+    assert.deepEqual(firstResult, { x: 1, y: 2 })
+    assert.strictEqual(selector({ a: 1, b: 2 }), firstResult)
+    const secondResult = selector({ a: 2, b: 2 })
+    assert.deepEqual(secondResult, { x: 2, y: 2 })
+    assert.strictEqual(selector({ a: 2, b: 2 }), secondResult)
+         */
+    }
+
+    @Ignore //not implemented
+    @Test
+    fun structuredSelectorWithCustomSelectorCreator() {
+        /*
+       const customSelectorCreator = createSelectorCreator(
+      defaultMemoize,
+      (a, b) => a === b
+    )
+    const selector = createStructuredSelector({
+      x: state => state.a,
+      y: state => state.b
+    }, customSelectorCreator)
+    const firstResult = selector({ a: 1, b: 2 })
+    assert.deepEqual(firstResult, { x: 1, y: 2 })
+    assert.strictEqual(selector({ a: 1, b: 2 }), firstResult)
+    assert.deepEqual(selector({ a: 2, b: 2 }), { x: 2, y: 2 })
+         */
+    }
+    @Test
+    fun resetComputationsTest() {
+        val selector = SelectorFor<StateA>()
+                .withField{ a}
+                .compute{a:Int -> a}
+
+        val state1= StateA(a=1)
+        assertThat(selector(state1)).isEqualTo(1)
+        assertThat(selector(state1)).isEqualTo(1)
+        assertThat(selector.recomputations).isEqualTo(1)
+        val state2 = StateA(a=2)
+        assertThat(selector(state2)).isEqualTo(2)
+        assertThat(selector.recomputations).isEqualTo(2)
+
+        selector.resetComputations()
+        assertThat(selector.recomputations).isEqualTo(0)
+
+        assertThat(selector(state1)).isEqualTo(1)
+        assertThat(selector(state1)).isEqualTo(1)
+        assertThat(selector.recomputations).isEqualTo(1)
+        assertThat(selector(state2)).isEqualTo(2)
+        assertThat(selector.recomputations).isEqualTo(2)
+    }
+    @Test
+    fun isChangedTest() {
+        val selector = SelectorFor<StateA>()
+                .withField { a}
+                .compute{a:Int -> a}
+        val state1= StateA(a=1)
+        assertThat(selector(state1)).isEqualTo(1)
+        assertThat(selector.isChanged()).isTrue()
+        selector.resetChanged()
+        assertThat(selector(state1)).isEqualTo(1)
+        assertThat(selector.isChanged()).isFalse()
+        val state2=StateA(a=2)
+        assertThat(selector(state2)).isEqualTo(2)
+        assertThat(selector.isChanged()).isTrue()
+    }
+
+    data class State3(val p1:Double, val p2:Double, val p3:Double)
+    @Test
+    fun args3Test() {
+        val selector = SelectorFor<State3>()
+                .withField { p1}
+                .withField { p2}
+                .withField { p3}
+                .compute{p1:Double,p2:Double,p3:Double -> p1/p2/p3}
+        val state= State3(1.0,2.0,3.0)
+        assertThat(selector(state)).isEqualTo(1.0/2.0/3.0)
+    }
+    data class State4(val p1:Double, val p2:Double, val p3:Double,val p4:Double)
+    @Test
+    fun args4Test() {
+        val selector = SelectorFor<State4>()
+                .withField { p1}
```

Comment:
Style: Might be a good idea to run a quick code style reformat on this file to pick up these small things. For example, `{ p1}` should be `{ p1 }`